crates/sui-json-rpc/src/read_api.rs

```diff
@@ -327,11 +331,13 @@ impl ReadApi {
                 };
                 results.push(get_balance_changes_from_effect(
                     &object_cache,
-                    resp.effects.as_ref().ok_or_else(|| {
-                        Error::SuiRpcInputError(SuiRpcInputError::GenericNotFound(
-                            "unable to derive balance changes because effect is empty".to_string(),
-                        ))
-                    })?,
+                    resp.effects
+                        .as_ref()
+                        .ok_or_else(|| ClientError::InvalidParam {
+                            param: "opts",
+                            reason: "unable to derive balance changes because effect is empty"
```

Comment:
how is this displayed? If a standalone string can we use sentence case? Unable to derive...and then . at the end?